app/lib/test_applications.rb

```diff
@@ -390,9 +390,15 @@ def make_offer(choice, conditions: ['Complete DBS'])
   def change_offer(choice, conditions: ['Complete DBS'])
     as_provider_user(choice) do
       fast_forward
-      year = choice.current_course.recruitment_cycle_year
-      new_course = choice.current_course.provider.courses
-                         .in_cycle(year).with_course_options.sample
+      current_course = choice.current_course
```

Comment:
maybe we can use `GetChangeOfferOptions` which can give us available courses instead?

Reply:
We could try that, but it doesn't just drop in like this. That query needs a provider user to be passed in to handle permissions. We'd have to be careful that provider permissions were set up properly. This way is a bit simpler and still fixes the bug.

WDYT?

Reply:
Very reasonable 👍